libsrc/stdlib/Lbsearch.asm

```diff
@@ -24,57 +24,42 @@ EXTERN l_jpix
 ;        carry set indicates item not found with HL=0
 ; uses : af, de, hl
 ;
-; Call Lbsearch + 3 if you prefer to enter with hl = last item in array
 
 .Lbsearch
-
-   dec hl
-   add hl,hl
-   add hl,de        ; HL = right side
-
-.loop               ; DE = left, HL = right
-   ld a,d           ; is left <= right or are we done?
-   cp h
-   jr c, slice
-   jr nz, done
-   ld a,e
-   cp l
-   jr nc, done
+.loop
+   ld a,h       ; Number of items still to check is 0? if so, return not found
+   or l
+   jr nz, slice
+   scf
+   ret
 
 .slice
-   push hl          ; stack right side
-   add hl,de        ; HL = left + right
-   rr h
-   rr l             ; HL = unrounded address of middle item
-
-   ld a,l           ; shenanigans to ensure HL aligns on item
-   xor e
-   rra
-   jr nc, compare
-   dec hl
+   push hl      ; Get the address at the halfway index + the base address
+   ld a, $fe
```

Comment:
is this for Z80? If yes, you can do also `res 0,l` to make HL even in 8T and two bytes.

Reply:
It will need to work on an 8080 as well as on a z80 I think

Reply:
Then again, the "rr h" instruction that was here previously isn't available on an 8080 either, so this function has never been 8080 friendly. I'll probably change the code to your suggestion to save a few T states. Every T state will count if this code is to search through large arrays

Reply:
https://github.com/z88dk/z88dk/wiki/Classic-8080#stdlibh - bsearch is not available on 808x

Reply:
This is a change to the l_bsearch function in stdlib. z88dk doesn't have bsearch at all it seems. The sample code I posted on the forums used a preprocessor macro to filter out the extra size parameter that bsearch has but that l_bsearch doesn't. bsearch on gcc can operate on arrays of pointers the same as l_bsearch (only) can.

Reply:
Ok @ped7g, I've run my tests and the opcode suggested works fine.